crates/vision/src/line_detection/checks.rs

```diff
@@ -0,0 +1,154 @@
+use std::ops::Range;
+
+use coordinate_systems::Pixel;
+use linear_algebra::{distance, vector, Point2, Vector2};
+use projection::{camera_matrix::CameraMatrix, Projection};
+use types::{image_segments::EdgeType, ycbcr422_image::YCbCr422Image};
+
+use super::segment::Segment;
+
+pub fn is_non_field_segment(segment: &Segment) -> bool {
+    segment.start_edge_type != EdgeType::Falling && segment.end_edge_type != EdgeType::Rising
+}
+
+pub fn is_in_length_range(
+    segment: &Segment,
+    camera_matrix: &CameraMatrix,
+    allowed_projected_segment_length: &Range<f32>,
+) -> bool {
+    let Ok(start) = camera_matrix.pixel_to_ground(segment.start.cast()) else {
+        return false;
+    };
+    let Ok(end) = camera_matrix.pixel_to_ground(segment.end.cast()) else {
+        return false;
+    };
+    allowed_projected_segment_length.contains(&distance(start, end))
+}
+
+pub fn is_maximum_segment(segment: &Segment) -> bool {
+    segment.start_edge_type == EdgeType::Rising && segment.end_edge_type == EdgeType::Falling
+}
```

Comment:
I think this isn't the same logic as on main.
Main returns true if the start edge is ***not*** rising or the end edge is ***not*** falling.

Reply:
I misunderstood the old code and implemented this as two checks. It should be improved now.